src/Uno.UI.Runtime.Skia.X11/X11XamlRootHost.cs

```diff
@@ -110,8 +110,8 @@ public X11XamlRootHost(X11WindowWrapper wrapper, Window winUIWindow, XamlRoot xa
 
 			if (_renderDirty)
 			{
-				_renderer?.Render();
 				_renderDirty = false;
+				_renderer?.Render();
```

Comment:
Does the order really matter here?

Reply:
It probably won't be noticeable, but there's a theoretical race condition where we during `_renderer?.Render()`, another thread might set `_renderDirty` to true and it that case it will be reset back to false after `Render()` ends, so we will lag a frame in that case.